src/pages/favorites/favorites.component.ts

```diff
@@ -164,6 +167,14 @@ export class FavoritesComponent {
   }
   deleteTrip(trip): void {
     _.remove(this.trips, {name: trip.name});
-    this.tripSvc.deleteTrip(trip);
+    this.faveTripSvc.deleteTrip(trip);
+  }
+  removeRoute(route): void {
+    _.remove(this.routes, {name: route.name});
```

Comment:
So, you're asking [Lodash](https://lodash.com/docs/4.17.4#remove) to `remove` entries in the `this.routes` array whose `name` property matches the `name` property of the route passed to this function.  

The problem here is that routes don't have a `name` property!  This line should be
`_.remove(this.routes, {RouteId: route.RouteId})`. 

The reason why the [favoriteRouteService function](https://github.com/umts/pvta-multiplatform/blob/master/src/providers/favorite-route.service.ts#L56) works is that it's properly removing based on `RouteId`.